### PR TITLE
[DOCS] a batch of edits for the configuration v2alpha1 file

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do?

Starting to edit and standardize these configs.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [ ] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [ ] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

**Review by kayayarai — APPROVED** [review]

File: docs/configuration.v2alpha1.md [hunk 1]
```diff
@@ -1,18 +1,28 @@
 # Configuration
 
+This page describes configuration options for Datadog Operator v1.0.0, which uses `DatadogAgent/v2alpha1`. For earlier versions of the Datadog Operator, see [configuration options for `DatadogAgent/v1alpha1`][7] or [migrate to `v2alpha1`][8].
```
Comment (levan-m) [inline docs/configuration.v2alpha1.md:3]: This whole file is generated. This section should be updated in the `header.markdown`
https://github.com/DataDog/datadog-operator/blob/main/hack/generate-docs/header.markdown

File: docs/configuration.v2alpha1.md [hunk 2]
```diff
@@ -374,3 +384,22 @@ In the table, `spec.override.nodeAgent.image.name` and `spec.override.nodeAgent.
 [3]: https://github.com/DataDog/datadog-operator/blob/main/examples/datadogagent/v2alpha1/datadog-agent-with-apm-hostport.yaml
 [4]: https://github.com/DataDog/datadog-operator/blob/main/examples/datadogagent/v2alpha1/datadog-agent-with-clusteragent.yaml
 [5]: https://github.com/DataDog/datadog-operator/blob/main/examples/datadogagent/v2alpha1/datadog-agent-with-tolerations.yaml
+[7]: https://github.com/DataDog/datadog-operator/blob/main/docs/configuration.v1alpha1.md
```
Comment (levan-m) [inline docs/configuration.v2alpha1.md:387]: This should be added to footer template here
https://github.com/DataDog/datadog-operator/blob/main/hack/generate-docs/v2alpha1_footer.markdown

File: docs/configuration.v2alpha1.md [hunk 3]
```diff
@@ -33,191 +43,191 @@ spec:
...
-| global.podLabelsAsTags | Provide a mapping of Kubernetes Labels to Datadog Tags. <KUBERNETES_LABEL>: <DATADOG_TAG_KEY> |
-| global.registry | Registry is the image registry to use for all Agent images. Use 'public.ecr.aws/datadog' for AWS ECR. Use 'docker.io/datadog' for DockerHub. Default: 'gcr.io/datadoghq' |
-| global.site | Site is the Datadog intake site Agent data are sent to. Set to 'datadoghq.com' to send data to the US1 site (default). Set to 'datadoghq.eu' to send data to the EU site. Set to 'us3.datadoghq.com' to send data to the US3 site. Set to 'us5.datadoghq.com' to send data to the US5 site. Set to 'ddog-gov.com' to send data to the US1-FED site. Set to 'ap1.datadoghq.com' to send data to the AP1 site. Default: 'datadoghq.com' |
-| global.tags | Tags contains a list of tags to attach to every metric, event and service check collected. Learn more about tagging: https://docs.datadoghq.com/tagging/ |
-| override | Override the default configurations of the agents |
+| `features.admissionController.agentCommunicationMode` | Corresponds to the mode used by the Datadog application libraries to communicate with the Agent. Values: `hostip`, `service`, or `socket`. |
```
Comment (levan-m) [inline docs/configuration.v2alpha1.md:46]: These are taken from godoc here
https://github.com/DataDog/datadog-operator/blob/main/apis/datadoghq/v2alpha1/datadogagent_types.go

**Review by levan-m — COMMENTED** [review]
Thanks for improving the documentation @cswatt! 
This file is generated and will be overwritten on next build. Added comments where changes should be made. One done, you run `make generate` and it should regenerate the file (assuming you have a working Go setup).

Maybe we should include a comment in header, footer so it appears in this file to avoid confusion in the future?

```
<!-- This file is generated, please make changes in... -->
```

**Comment by cswatt** [comment]
@levan-m Thanks for the heads up! I'll close this PR and make my changes in the correct files :)

**Pull request closed without merging**